client/python/transaction.py

```diff
@@ -0,0 +1,81 @@
+import time
+
+from cache_map import *
+from client_cache import *
+from tx_clock import *
+
+class Transaction(object):
+    def __init__(self, cache, read_timestamp=None, max_age=None, no_cache=False):
+        if (type(cache) != ClientCache):
```

Comment:
I don't think you want to enforce this. In fact, I think you specifically want to pass in a mock of ClientCache for testing transaction.